app/post.go

```diff
@@ -586,6 +596,45 @@ func (a *App) GetPostsAroundPost(postId, channelId string, offset, limit int, be
 	}
 }
 
+func (a *App) GetPostsForChannelAroundLastUnread(channelId, userId string) (*model.PostList, *model.AppError) {
+	var member *model.ChannelMember
+	var err *model.AppError
+	if member, err = a.GetChannelMember(channelId, userId); err != nil {
```

Comment:
Is it possible to make any of these database calls run in parallel?

Reply:
Would be good but I suggest to initially start this way to conveniently use the existing `GetPostSince` of `sqlstore` and `app` packages.

Reply:
I do get the benefit of runnin in parallel, why your suggestion to initially start this way?

Reply:
I foresee this API to be called only once maybe during initial view of a channel.  It's not frequently used.  Anyway, I'll update

Reply:
I tried to make parallel queries but I've got two issues:
- duplicated select statement [[1](https://github.com/mattermost/mattermost-server/pull/9108/files#diff-3839c6aa88b8e4cfee8869d3da6d20adR636), [2](https://github.com/mattermost/mattermost-server/pull/9108/files#diff-3839c6aa88b8e4cfee8869d3da6d20adR652)] in getting `LastViewedAt` of ChannelMember.  Option might be to set SQL variable but it might introduce race condition depending on how it's used per DB session. In addition to that, MySQL and PostgreSQL have different implementation.  Actually, i've got that idea from @jespino out of our offline discussion.
- complicated to compare `LastViewedAt` of new channel member since the default value is set to 0.  For new channel member (added/joined), this API should return the last recent posts as if all previous posts were already read by the user.  However, if the value of `LastViewedAt` is `0`, then it will return the old posts of a channel which were posted before the user became a member.  By using separate query in getting channel member, we can easily compare it and exit immediately if that is 0.

My proposal is to not run it in parallel.  That is to get the channel member and then get posts since using existing queries.  If you'll agree, I'll revert the latest change I've made.